### ISSUE TITLE
Nested local labels

### ISSUE THREAD

**Issue by fmafma** [z00m128/sjasmplus#244]
How hard would it be to implement nested local labels?

The idea is to be able to write:

```
label1

.local1  ; label1.local1

..next  ; label1.local1.next

...loop1  ; label1.local1.next.loop1

...loop2  ; label1.local1.next.loop2

.local2  ; label1.local2

..next  ; label1.local2.next

...loop1  ; label1.local2.next.loop1
```

Each additional dot refers to the previous local label containing 1 dot less...

Thanks!

**Comment by ped7g** [comment]
maybe between easy and medium, can't think of any difficult issue with this.

But I don't instantly like this, feels way too fancy/abundant.

But let's keep it open here and see if there's more people with opinion on it, maybe showing it more favor than me (at least I'm not hard against it, it's not mandatory to use it if it would be added).

**Comment by fmafma** [comment]
Thanks!

It could be useful to organize defines:

```
OBJ
.POS
..X     EQU 100
..Y     EQU 150
.DIR
..X     EQU 1
..Y     EQU 2
.COLOR
..FOR   EQU WHITE
..BACK  EQU BLACK

        LD   H,OBJ.POS.X
        LD   L,OBJ.POS.Y
        LD   D,OBJ.DIR.X
        LD   E,OBJ.DIR.Y
        LD   B,OBJ.COLOR.FOR
        LD   C,OBJ.COLOR.BACK
        ; Do something
```
or in sub-routines:
```
mySubRoutine
        ; Some code
.part1
        ; Some code
..loop
        JR   NZ,.next
        DJNZ .loop
..next
        ; Some code

.part2
        ; Some code
..loop
        JR   NZ,.next
        DJNZ .loop
..next
        ; Some code
```

**Comment by fmafma** [comment]
I agree that the use case in the sub-routine is not very good¹, but I'm not a big fan to introduce a new paradgim (with 2 more keywords!) to implement the define hierarchy; for this use case, I think nested labels are very easy to use and understand.

For the maintenance of SJASM+, on the other hand, I can't say. Hence my initial question ;o) Maybe something to think about for branch 2.x?

Best,

¹ even if jumps in a long code is always complicated to read, nested labels or not; long routines are never a good idea, whatever language you use.

**Comment by fmafma** [comment]
In fact, I would definitly have great use of nested labels in sub-routines!

Even in a reasonnable size one, I often have several sub-sections, with their own loops/relative jumps.
Without nested labels, I need to increment the names (loop1, loop2... or next1, next2...), and have them renamed when adding/removing code.
With nested labels, it would be much easier.
